bitnami/thanos/templates/storegateway/statefulset-sharded.yaml

```
@@ -0,0 +1,257 @@
+{{- if and .Values.storegateway.enabled .Values.storegateway.sharded.enabled }}
+
+{{- $shards := int 0 }}
+{{- $hashPartioning := false }}
+{{- $timePartioning := false }}
+
+{{- if .Values.storegateway.sharded.hashPartioning.shards }}
+  {{- $shards = int .Values.storegateway.sharded.hashPartioning.shards }}
+  {{- $hashPartioning = true }}
+{{- else }}
+  {{- $shards = len .Values.storegateway.sharded.timePartioning }}
+  {{- $timePartioning = true }}
+{{- end }}
+
+{{- range $index, $_ := until $shards }}
+apiVersion: {{ include "common.capabilities.statefulset.apiVersion" $ }}
+kind: StatefulSet
+metadata:
+  name: {{ include "common.names.fullname" $ }}-storegateway-{{ $index }}
```

We should truncate this to avoid exceeding the 63 characters DNS limit:
```suggestion
  name: {{ printf "%s-storegateway-%s" (include "common.names.fullname" $) $index | trunc 63 | trimSuffix "-" }}
```